src/main/java/com/commercetools/sync/products/helpers/VariantReferenceResolver.java

```diff
@@ -162,19 +175,27 @@ static boolean isProductReference(@Nonnull final JsonNode referenceValue) {
             .orElse(false);
     }
 
+    private static boolean isCategoryReference(@Nonnull final JsonNode referenceValue) {
+        return getReferenceTypeIdIfReference(referenceValue)
+            .map(referenceTypeId -> Objects.equals(referenceTypeId, Category.referenceTypeId()))
+            .orElse(false);
+    }
+
     @Nonnull
     private static Optional<String> getReferenceTypeIdIfReference(@Nonnull final JsonNode referenceValue) {
         final JsonNode typeId = referenceValue.get(REFERENCE_TYPE_ID_FIELD);
         return Optional.ofNullable(typeId).map(JsonNode::asText);
     }
 
     @Nonnull
-    CompletionStage<Optional<String>> getProductResolvedIdFromKeyInReference(@Nonnull final JsonNode referenceValue) {
+    private CompletionStage<Optional<String>> getResolvedIdFromKeyInReference(
+        @Nonnull final JsonNode referenceValue,
+        @Nonnull final Function<String, CompletionStage<Optional<String>>> cacheFetcher) {
```

Comment:
I would use `resolvedIdFetcher` instead of `cacheFetcher` because the cache is only an implementation.

Reply:
completely agree.

Reply:
828d488